### PR TITLE
Added support for Grape 0.10.x

### LINKED ISSUE #35: Issue with Grape: undefined local variable or method `settings' for #<Grape::Endpoint>
I encountered the following error when using `grape-active_model_serializers 1.2.0` with `grape 0.10.1`. I tried version `1.3.1` and the issue still exists.

    undefined local variable or method `settings' for #<Grape::Endpoint:0x007f9ed79dd690>

	grape-active_model_serializers (1.2.0) lib/grape-active_model_serializers/endpoint_extension.rb:12:in `namespace_options'
	grape-active_model_serializers (1.2.0) lib/grape-active_model_serializers/formatter.rb:27:in `build_options_from_endpoint'
	grape-active_model_serializers (1.2.0) lib/grape-active_model_serializers/formatter.rb:17:in `fetch_serializer'
	grape-active_model_serializers (1.2.0) lib/grape-active_model_serializers/formatter.rb:6:in `call'
	grape (0.10.1) lib/grape/middleware/formatter.rb:33:in `block in after'
	grape (0.10.1) lib/grape/middleware/formatter.rb:32:in `collect'
	grape (0.10.1) lib/grape/middleware/formatter.rb:32:in `after'
	grape (0.10.1) lib/grape/middleware/base.rb:25:in `call!'
	grape (0.10.1) lib/grape/middleware/base.rb:18:in `call'
	grape (0.10.1) lib/grape/middleware/base.rb:24:in `call!'
	grape (0.10.1) lib/grape/middleware/base.rb:18:in `call'
	grape (0.10.1) lib/grape/middleware/error.rb:27:in `block in call!'
	grape (0.10.1) lib/grape/middleware/error.rb:26:in `catch'
	grape (0.10.1) lib/grape/middleware/error.rb:26:in `call!'
	grape (0.10.1) lib/grape/middleware/base.rb:18:in `call'
	rack (1.5.2) lib/rack/head.rb:11:in `call'
	rack (1.5.2) lib/rack/builder.rb:138:in `call'
	grape (0.10.1) lib/grape/endpoint.rb:196:in `call!'
	grape (0.10.1) lib/grape/endpoint.rb:184:in `call'
	rack-mount (0.8.3) lib/rack/mount/route_set.rb:152:in `block in call'
	rack-mount (0.8.3) lib/rack/mount/code_generation.rb:96:in `block in recognize'
	rack-mount (0.8.3) lib/rack/mount/code_generation.rb:75:in `optimized_each'
	rack-mount (0.8.3) lib/rack/mount/code_generation.rb:95:in `recognize'
	rack-mount (0.8.3) lib/rack/mount/route_set.rb:141:in `call'
	grape (0.10.1) lib/grape/api.rb:102:in `call'
	grape (0.10.1) lib/grape/api.rb:33:in `call!'
	grape (0.10.1) lib/grape/api.rb:29:in `call'
	actionpack (4.1.8) lib/action_dispatch/journey/router.rb:73:in `block in call'
	actionpack (4.1.8) lib/action_dispatch/journey/router.rb:59:in `each'
	actionpack (4.1.8) lib/action_dispatch/journey/router.rb:59:in `call'
	actionpack (4.1.8) lib/action_dispatch/routing/route_set.rb:678:in `call'
	warden (1.2.3) lib/warden/manager.rb:35:in `block in call'
	warden (1.2.3) lib/warden/manager.rb:34:in `catch'
	warden (1.2.3) lib/warden/manager.rb:34:in `call'
	rack (1.5.2) lib/rack/etag.rb:23:in `call'
	rack (1.5.2) lib/rack/conditionalget.rb:25:in `call'
	rack (1.5.2) lib/rack/head.rb:11:in `call'
	actionpack (4.1.8) lib/action_dispatch/middleware/params_parser.rb:27:in `call'
	actionpack (4.1.8) lib/action_dispatch/middleware/flash.rb:254:in `call'
	rack (1.5.2) lib/rack/session/abstract/id.rb:225:in `context'
	rack (1.5.2) lib/rack/session/abstract/id.rb:220:in `call'
	actionpack (4.1.8) lib/action_dispatch/middleware/cookies.rb:560:in `call'
	activerecord (4.1.8) lib/active_record/query_cache.rb:36:in `call'
	activerecord (4.1.8) lib/active_record/connection_adapters/abstract/connection_pool.rb:621:in `call'
	activerecord (4.1.8) lib/active_record/migration.rb:380:in `call'
	actionpack (4.1.8) lib/action_dispatch/middleware/callbacks.rb:29:in `block in call'
	activesupport (4.1.8) lib/active_support/callbacks.rb:82:in `run_callbacks'
	actionpack (4.1.8) lib/action_dispatch/middleware/callbacks.rb:27:in `call'
	actionpack (4.1.8) lib/action_dispatch/middleware/reloader.rb:73:in `call'
	actionpack (4.1.8) lib/action_dispatch/middleware/remote_ip.rb:76:in `call'
	actionpack (4.1.8) lib/action_dispatch/middleware/debug_exceptions.rb:17:in `call'
	actionpack (4.1.8) lib/action_dispatch/middleware/show_exceptions.rb:30:in `call'
	railties (4.1.8) lib/rails/rack/logger.rb:38:in `call_app'
	railties (4.1.8) lib/rails/rack/logger.rb:20:in `block in call'
	activesupport (4.1.8) lib/active_support/tagged_logging.rb:68:in `block in tagged'
	activesupport (4.1.8) lib/active_support/tagged_logging.rb:26:in `tagged'
	activesupport (4.1.8) lib/active_support/tagged_logging.rb:68:in `tagged'
	railties (4.1.8) lib/rails/rack/logger.rb:20:in `call'
	rack (1.5.2) lib/rack/methodoverride.rb:21:in `call'
	activesupport (4.1.8) lib/active_support/cache/strategy/local_cache_middleware.rb:26:in `call'
	rack (1.5.2) lib/rack/lock.rb:17:in `call'
	actionpack (4.1.8) lib/action_dispatch/middleware/static.rb:84:in `call'
	railties (4.1.8) lib/rails/engine.rb:514:in `call'
	railties (4.1.8) lib/rails/application.rb:144:in `call'
	newrelic_rpm (3.9.8.273) lib/new_relic/rack/developer_mode.rb:56:in `traced_call'
	newrelic_rpm (3.9.8.273) lib/new_relic/agent/instrumentation/middleware_tracing.rb:55:in `call'
	rack (1.5.2) lib/rack/lock.rb:17:in `call'
	rack (1.5.2) lib/rack/content_length.rb:14:in `call'
	rack (1.5.2) lib/rack/handler/webrick.rb:60:in `service'

### PR DESCRIPTION
**Pull request by dblock**
Fixes #35, adds tests for all versions of Grape on Travis, see https://travis-ci.org/dblock/grape-active_model_serializers/builds/46861142 for an example.

### CODE REVIEW

File: lib/grape-active_model_serializers/formatter.rb [hunk 1]
```diff
@@ -26,21 +26,21 @@ def fetch_serializer(resource, env)
...
         def other_options
           options = {}
-          if @meta_content_items
-            if @meta_key
-             key_option = @meta_key[:meta_key]
-              @meta_key.delete(:meta_key)
+          if Formatter::ActiveModelSerializers.meta
+            if Formatter::ActiveModelSerializers.meta_key
+              key_option = Formatter::ActiveModelSerializers.meta_key[:meta_key]
+              Formatter::ActiveModelSerializers.meta_key.delete(:meta_key)
```
Comment (siong1987) [inline lib/grape-active_model_serializers/formatter.rb]: we can just do `key_option = Formatter::ActiveModelSerializers.meta_key.delete(:meta_key)` here.

File: lib/grape-active_model_serializers/formatter.rb [hunk 2]
```diff
@@ -26,21 +26,21 @@ def fetch_serializer(resource, env)
...
               options[:meta_key] = key_option if key_option
             end
-            meta_option = @meta_content_items[:meta]
-            @meta_content_items.delete(:meta)
+            meta_option = Formatter::ActiveModelSerializers.meta[:meta]
+            Formatter::ActiveModelSerializers.meta.delete(:meta)
```
Comment (siong1987) [inline lib/grape-active_model_serializers/formatter.rb]: same thing here.

File: lib/grape-active_model_serializers/formatter.rb [hunk 3]
```diff
@@ -62,12 +62,20 @@ def build_options_from_endpoint(endpoint)
...
+              endpoint.options[:path][0].to_s.split('/')[-1]
+            end
           else
-            endpoint.options[:path][0].to_s.split('/')[-1]
+            innermost_scope = endpoint.settings.peek
+            if innermost_scope[:namespace]
+              innermost_scope[:namespace].space
+            else
+              endpoint.options[:path][0].to_s.split('/')[-1]
+            end
```
Comment (siong1987) [inline lib/grape-active_model_serializers/formatter.rb]: we can do:

```
innermost_scope = if endpoint.respond_to?(:namespace_stackable)
  endpoint.namespace_stackable(:namespace).last
else
  endpoint.settings.peek[:namespace]
end

if innermost_scope
  innermost_scope.space
else
  endpoint.options[:path][0].to_s.split('/')[-1]
end
```

**Comment by dblock** [comment]
Updated, simpler.

**Comment by dblock** [comment]
I have another change in https://github.com/dblock/grape-active_model_serializers/tree/one-meta, will PR next, where we pass meta around with `env`, which is probably the right thing to do - we should not be assigning globals that belong to modules, which I suspect may cause leaks or thread-safety issues on some platforms.

**Comment by siong1987** [comment]
@dblock that is awesome!  thanks!

**Pull request merged**